### PR TITLE
Add orca spark pytorch estimator wrapper

### PR DESCRIPTION
**Pull request by cyita**

### CODE REVIEW

File: pyzoo/zoo/orca/learn/pytorch/estimator.py [hunk 1]
```diff
@@ -15,6 +15,9 @@
 #
 from zoo.orca.learn.pytorch.training_operator import TrainingOperator
 from zoo.orca.learn.pytorch.pytorch_horovod_estimator import PyTorchHorovodEstimator
+from zoo.pipeline.estimator.estimator import Estimator as SparkEstimator
+from bigdl.optim.optimizer import MaxEpoch
+from zoo.feature.common import FeatureSet
 
 
 class Estimator(object):
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/pytorch/estimator.py:23]: change to `from_torch`

File: pyzoo/zoo/orca/learn/pytorch/estimator.py [hunk 2]
```diff
@@ -38,15 +41,23 @@ def from_model_creator(*,
...
-        assert backend == "ray", "only ray backend is supported for now"
-        return PyTorchHorovodEstimatorWrapper(model_creator=model_creator,
-                                              optimizer_creator=optimizer_creator,
-                                              loss_creator=loss_creator,
-                                              scheduler_creator=scheduler_creator,
-                                              training_operator_cls=training_operator_cls,
-                                              initialization_hook=initialization_hook,
-                                              config=config,
-                                              scheduler_step_freq=scheduler_step_freq)
+        if backend == "ray":
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/pytorch/estimator.py:44]: change to horovod vs. bigdl

File: pyzoo/zoo/orca/learn/pytorch/estimator.py [hunk 3]
```diff
@@ -37,16 +42,24 @@ def from_torch(*,
...
+                                                  config=config,
+                                                  scheduler_step_freq=scheduler_step_freq)
+        elif backend == "bigdl":
+            return PytorchSparkEstimatorWrapper(model=model_creator,
+                                                loss=loss_creator,
+                                                optimizer=optimizer_creator,
+                                                model_dir=None,
+                                                bigdl_type="float")
+        else:
+            raise ValueError("only horovod and bigdl backend are supported for now")
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/pytorch/estimator.py:62]: change the `model_creator`, `loss_creator`, `optimizer_creator` arguments to `model`, `loss`, `optimizer`

File: pyzoo/zoo/orca/learn/pytorch/estimator.py [hunk 4]
```diff
@@ -37,16 +42,24 @@ def from_torch(*,
...
-                                              optimizer_creator=optimizer_creator,
-                                              loss_creator=loss_creator,
-                                              scheduler_creator=scheduler_creator,
-                                              training_operator_cls=training_operator_cls,
-                                              initialization_hook=initialization_hook,
-                                              config=config,
-                                              scheduler_step_freq=scheduler_step_freq)
+                   backend="horovod"):
+        if backend == "horovod":
+            return PyTorchHorovodEstimatorWrapper(model_creator=model_creator,
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/pytorch/estimator.py:47]: does horovod support `model_dir` @yangw1234 ?

**Comment by cyita** [comment]
jenkins passed: http://10.239.47.210:18888/view/ZOO-PR-PIPELINE/job/ZOO-PR-Validation/4002/

File: pyzoo/zoo/orca/learn/pytorch/utils.py [hunk 5]
```diff
@@ -258,3 +258,18 @@ def overrider(method):
         return method
 
     return overrider
+
+
+# todo: this might be very slow
+def to_sample(data):
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/pytorch/utils.py:264]: we should move non-pytorch and non-tf specific utils to a common place (such as `orca.data.utils` or `orca.learn.utils`) maybe in a follow-up PR.

**Comment by cyita** [comment]
jenkins jep ut passed: http://10.239.47.210:18888/job/ZOO-PR-Pip-JEP-UT/676/console

**Pull request merged**